### PR TITLE
Rain gauge

### PR DESCRIPTION
**Pull request by gerard33**
Added a rain gauge

rain gauge

rain gauge 2

### CODE REVIEW

File: src/widgets/DeviceWidget.js [hunk 1]
```diff
@@ -130,6 +131,11 @@ class DeviceWidget extends Component {
             direction={device.svalue2}
             speed={Number(device.svalue3)}
             {...this.props} />
+      case "Rain" :
+        return <RainWidget idx={device.idx} label={device.name}
```
Comment (t0mg) [inline src/widgets/DeviceWidget.js]: FYI, idx is not necessary in widget (read only) types.

File: src/widgets/RainWidget.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+import './RainWidget.css';
+
+class RainWidget extends Component {
+
+  render() {
+    return (
+      <GenericWidget class="RainWidget" icon="cloud"
+          isOn={this.props.rain > 0}
+          value1={<div className="rain">{this.props.rain}</div>}
+          value2={<div className="rate">{this.props.rate}</div>}
```
Comment (t0mg) [inline src/widgets/RainWidget.js]: You don't need extra divs here, see for example WattWidget. In the css you can simply do:

.RainWidget h2:after {
  content: ' mm';
}

.RainWidget .value2:after {
  content: ' mm/h';
}

File: src/widgets/DeviceWidget.js (hunk 1 above)
Comment (gerard33) [reply src/widgets/DeviceWidget.js]: Removed the idx part

File: src/widgets/RainWidget.js (hunk 2 above)
Comment (gerard33) [reply src/widgets/RainWidget.js]: Thanks, I indeed used the WattWidget method, but I used value1 in stead of h2, so it didn't work for me. But this works :)

Comment (t0mg) [reply src/widgets/RainWidget.js]: Yeah this was poor design, I should add a value1 class to the h2. Thanks for the update.

Comment (t0mg) [reply src/widgets/RainWidget.js]: BTW I realize that we're mixin css2 (:after) and css3 (::after) notation. Both are accepted but it's a little messy. I'll do a pass on the codebase to use ::after everywhere.

**Pull request merged**